### PR TITLE
ICU-21730 Port `testGetPreferencesFor` ICU4C tests to ICU4J

### PR DESCRIPTION
**Pull request by younies**
closes: https://github.com/icu-units/icu/issues/129

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21730
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [ ] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by younies** [comment]
@sffc , testGetAllConversionRates it does not make sense, because of this I have deleted it from C++.

File: icu4c/source/test/intltest/units_data_test.cpp [hunk 1]
```diff
@@ -60,21 +58,6 @@ void UnitsDataTest::testGetUnitCategory() {
     }
 }
 
-void UnitsDataTest::testGetAllConversionRates() {
```
Comment (sffc) [inline icu4c/source/test/intltest/units_data_test.cpp:63]: This looks like a data sanity test.  I don't know exactly why this test existed, but I think we should probably keep it.  However, we don't need to port it to Java.  It's okay to have tests like this one in C++ but not in Java.

**Review by sffc — COMMENTED** [review]
@hugovdm also mentioned `testGetUnitCategory` in https://github.com/icu-units/icu/issues/129.  Do we need to port that one as well?

File: icu4c/source/test/intltest/units_data_test.cpp (hunk 1 above)
Comment (younies) [reply icu4c/source/test/intltest/units_data_test.cpp:63]: @sffc , `testGetUnitCategory` is added in this PR: https://github.com/unicode-org/icu/pull/1839

For `testGetAllConversionRates`, we already testing this in testing the `UnitsConverter`, and the test should be always for the public API, except we need to make more deep testing.

Comment (sffc) [reply icu4c/source/test/intltest/units_data_test.cpp:63]: I prefer that we keep the test only in C++ and add a comment explaining that.

**Review by sffc — CHANGES_REQUESTED** [review]
Please restore the deleted test, but LGTM otherwise

**Review by sffc — APPROVED** [review]

**Pull request merged**